[woowahan1] Step2 구현

Jeju-Boobby opened the pull request.

smartyjpark committed:
step2 init

Jeju-Boobby committed:
[Added] 노말프레임, 볼 테스트 추가

Jeju-Boobby committed:
[Added] FinalFrame Test 추가 및 테스트 페키지 정리

smartyjpark committed:
Added reduce logic in NormalFrame Class

kkiluk committed:
[ADD] FinalFrame 점수 계산 기능을 구현하다.

FinalFrameTest 클래스에서 각각의 경우에 따른 점수 계산 테스트코드를 만들고,
FinalFrame 클래스에서 이를 구현했습니다. (TDD!!)

kkiluk committed:
[ADD] 누적점수 출력 기능을 구현하다.

각 프레임의 누적점수를 보여주는 기능을 구현했습니다.
스트라이크, 스페어 처리 등으로 현재 프레임의 점수가 확정되지 않은 경우에는 점수를 보여주지 않습니다. (so, 완벽!)

Jeju-Boobby committed:
[ADD] View 및 GameManager 리펙토링

View에 들어가있던 Frame에 종속되어있던 부분을 GameManager로 추출
GameManager 의 메소드 분리

javajigi commented on src/main/java/controller/GameManager.java:
```diff
@@ -4,27 +4,89 @@
+
+    private void runFrame(Frame currentFrame) {
+        currentFrame.bowl(ConsoleView.askPinCount(currentFrame.getRound()));
+        ConsoleView.printScoreBoard(playerName, frameStatusToString(), frameScoreToString());
```
볼링 점수 판에 출력할 status와 score를 인스턴스 변수로 가지는 FrameResult와 같은 클래스를 추가한다.
메소드를 두번 나누어 접근하기보다 `List<FrameResult> getFrameResults()`와 같은 구조로 구현해 본다.

javajigi reviewed (changes requested):
짧은 시간 동안 구현하느라 고생했네요. 내일은 step2의 힌트, javajigi 브랜치에 제가 구현한 코드 리뷰해 보고 리팩토링 진행해 보세요.

kkiluk committed:
[REFACTORING] 프레임 상태와 누적점수를 포괄하는 FrameResult를 생성하다.

각각의 리스트로 존재하던 프레임의 상태와 누적점수를 하나의 DTO 클래스인 FrameResult에서 관리하도록 리팩토링했습니다.
따라서 GameManager에 있던 frameStatusToString과 frameScoreToString 메소드에서 처리하던 기능을 createFrameResults 메소드로 단일화했습니다.

smartyjpark committed:
Refactoring: Rename Class and Method / added varibles about max/min count of pins
